### PR TITLE
SLi-GRU + Li-GRU fixes

### PR DESCRIPTION
**Pull request by Adel-Moumen**

### CODE REVIEW

File: speechbrain/nnet/RNN.py [hunk 1]
```diff
@@ -1269,7 +1292,412 @@ def _ligru_cell(self, w, ht):
...
+    Arguments
+    ---------
+    hidden_size : int
+        Number of output neurons (i.e, the dimensionality of the output).
+        values (i.e, time and frequency kernel sizes respectively).
+    input_shape : tuple
+        The shape of an example input.
+    nonlinearity : str
+        Type of nonlinearity (tanh, relu).
+    ff_normalization : str
```
Comment (AsuMagic) [inline speechbrain/nnet/RNN.py:1389]: The paper doesn't mention (often?) the feedforward LN/BN so IMO, a reader could easily confuse SLi-GRU's LN in the recurrent layer with the feedforward LN/BN and believe that it refers to the LN. It might be worth mentioning something among the lines of "Note that a LayerNorm is applied in the recurrent network regardless of normalization applied on the feed-forward network"?

**Review by AsuMagic — CHANGES_REQUESTED** [review]

File: speechbrain/nnet/RNN.py (hunk 1 above)
Comment (Adel-Moumen) [reply speechbrain/nnet/RNN.py:1389]: Yes we could do that.

File: speechbrain/nnet/RNN.py [hunk 2]
```diff
@@ -1123,26 +1131,28 @@ def _forward_ligru(self, x, hx: Optional[Tensor]):
...
 
     Arguments
     ---------
     input_size : int
         Feature dimensionality of the input tensors.
-    batch_size : int
-        Batch size of the input tensors.
     hidden_size : int
         Number of output neurons.
     num_layers : int
```
[the remark this answers is not in the document]
Comment (AsuMagic) [reply speechbrain/nnet/RNN.py:1142]: I meant `num_layers` not `batch_size`, I should have clarified :)

File: speechbrain/nnet/RNN.py [hunk 3]
```diff
@@ -1190,11 +1202,18 @@ def __init__(
...
 
+        # we freeze the bias of the normalization layer
+        if not self.bias:
+            self.norm.bias.data.fill_(0)
+            self.norm.bias.requires_grad = False
+
         # Initial state
-        self.register_buffer("h_init", torch.zeros(1, self.hidden_size))
+        self.register_buffer(
+            "h_init", torch.zeros(self.batch_size, self.hidden_size)
```
[the remark this answers is not in the document]
Comment (AsuMagic) [reply speechbrain/nnet/RNN.py]: Done

File: speechbrain/nnet/RNN.py (hunk 1 above)
Comment (AsuMagic) [reply speechbrain/nnet/RNN.py:1389]: Done

File: speechbrain/nnet/RNN.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (AsuMagic) [reply speechbrain/nnet/RNN.py:1142]: Non-issue, can be marked resolved

**Comment by AsuMagic** [comment]
Rebased against tip of `develop`

**Comment by AsuMagic** [comment]
For the record, this is not ready to merge as the LS test recipe does not seem to converge for me (Adel and I are trying to figure that one out).
